infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SnapshotSender.java

```diff
@@ -106,6 +106,7 @@ public void transmit(UUID snapshotSyncEventId) {
         int messagesSent = 0;       // Limit the number of messages to maxNumSnapshotMsgPerBatch. The reason we need to limit
         // is because by design several state machines can share the same thread pool,
         // therefore, we need to hand the thread for other workers to execute.
+        long messagesSentSize = 0;
```

Comment:
This method can get invoked multiple times during a snapshot sync.  This variable cannot get reset here.  Can you reset it in the reset() method?